src/js/requestHandlers.js

```diff
@@ -230,6 +234,32 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (storyContent, uploadedFilesHandlerPath) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        if (block.blockType === "image") {
+            filesToDelete.push(block.imageUrl);
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            filesToDelete.push(block.mediaUrl);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        var filePath = "./" + uploadedFilesHandlerPath + "/" + fileToDelete;
```

Comment:
Thinking about options here.

We could use [`path.basename`](https://nodejs.org/api/path.html#path_path_basename_path_ext) prior to constructing the path to operate on.

But I'm thinking that instead we do a whitelist test (say using a regular expression) that the media URL contains only exactly the characters that we expect -- we construct the names used here, so we can reliably test. I think this would be:

- upper and lower case letters (unless we normalize the extension to lower case?)
- numbers
- "-"
- "."

Reply:
Let's add a new function `isValidMediaFilename` that checks filename contents as above and can also test for other cases such as `null`, `undefined`, empty string, and such. And if it's a function, we can unit test it.

Reply:
Excellent catch! I've added checks to ensure:
- the filename is truthy (covers `null`, `undefined`, empty string, etc.)
- the filename is a valid UUID (all stored files should be)
- the file can be accessed for read and write in the uploads directory